v2/yarpcgrpc/handler.go

```diff
@@ -246,3 +246,32 @@ func handlerErrorToGRPCError(err error, mdWriter *metadataWriter) error {
 	}
 	return status.Error(grpcCode, message)
 }
+
+func handleResponse(
+	encoding yarpc.Encoding,
+	errInfo *yarpcerror.Info,
+	resBuf *yarpc.Buffer,
+	serverStream grpc.ServerStream,
+	mdWriter *metadataWriter,
+) error {
+	// This is a regular response that we should send
+	if errInfo == nil && resBuf != nil {
+		return serverStream.SendMsg(resBuf.Bytes())
+	}
+
+	// This is an application error
+	if errInfo != nil {
+		if errInfo.Name != "" {
+			mdWriter.AddSystemHeader(ErrorNameHeader, errInfo.Name)
+		}
+		if resBuf != nil {
```

Comment:
this should never be `nil`, correct?

Reply:
not necessarily; take json for example: if they specify a yarpcerror with no details in their user handler, we don't put anything in the buffer.

we could make it always have a non-nil value by putting an empty buffer there though.